pkgs/by-name/no/noi/package.nix

```diff
@@ -0,0 +1,30 @@
+{
+  lib,
+  appimageTools,
+  fetchurl,
+  ...
+}:
+appimageTools.wrapType2 rec {
+  pname = "noi";
+  version = "0.4.0";
+  src = fetchurl {
+    url = "https://github.com/lencx/Noi/releases/download/v${version}/Noi_linux_${version}.Appimage";
+    hash = "sha256-ZwI1MpEoQn48zaan/GB7St6b15jtPHjwoUfD6bPkA3A=";
+  };
+
+  extraInstallCommands =
+    let
+      contents = appimageTools.extract { inherit pname version src; };
+    in
+    ''
+      install -m 444 -D ${contents}/Noi.desktop -t $out/share/applications
+    '';
+
+  meta = {
+    description = "AI-enhanced, customizable browser designed to streamline your digital experience";
+    homepage = "https://noi.nofwl.com";
+    maintainers = with lib.maintainers; [ lurgrid ];
+    license = lib.licenses.unfree;
```

Comment:
Can you add the line `sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];` ?